### ISSUE TITLE
1.2.5 now prefixing devices with "_ucm0001."

### ISSUE THREAD

**Issue by craftyguy** [alsa-project/alsa-ucm-conf#104]
I'm using a ucm2 config that is not yet upstream (see [1]), and 1e6297b650114cb2e043be4c677118f971e31eb7 has caused devices to get renamed; they are now prefixed with `_ucm0001.`:

alsaucm json output from 1.2.4:

```
...
  "Verbs": {
    "HiFi": {
      "Comment": "Default",
      "Devices": {
        "Handset": {
          "Comment": "Handset",
          "ConflictingDevices": [
            "Speaker",
            "Headphones"
          ],
          "Values": {
            "CaptureCTL": "hw:L5",
            "PlaybackCTL": "hw:L5",
            "PlaybackChannels": "2",
            "PlaybackPCM": "hw:L5,0",
            "PlaybackPriority": "100",
            "PlaybackSwitch": "name='Speaker Switch'",
            "PlaybackVolume": "name='Speaker Volume'"
          }
        },
...
```

alsaucm json output from 1.2.5:

```
...
  "Verbs": {
    "HiFi": {
      "Comment": "Default",
      "Devices": {
        "Handset": {
          "Comment": "Handset",
          "ConflictingDevices": [
            "Speaker",
            "Headphones"
          ],
          "Values": {
            "CaptureCTL": "_ucm0001.hw:L5",
            "PlaybackCTL": "_ucm0001.hw:L5",
            "PlaybackChannels": "2",
            "PlaybackPCM": "_ucm0001.hw:L5,0",
            "PlaybackPriority": "100",
            "PlaybackSwitch": "name='Speaker Switch'",
            "PlaybackVolume": "name='Speaker Volume'"
          }
        },
...
```

This breaks things [like pulseaudio config](https://source.puri.sm/Librem5/librem5-base/-/blob/pureos/byzantium/default/audio/pulse/librem5.pa) that expects specific device names. Was this renaming intentional?

cc @agx

1. https://source.puri.sm/Librem5/librem5-base/-/tree/pureos/byzantium/default/audio/ucm2/Librem_5

**Comment by craftyguy** [comment]
If I remove the include added in 1e6297b, device names go back to what they were <1.2.5:

```
diff --git a/ucm2/ucm.conf b/ucm2/ucm.conf
index a9bdd0d..157742e 100644
--- a/ucm2/ucm.conf
+++ b/ucm2/ucm.conf
@@ -18,7 +18,7 @@ Define.V2ConfD yes    # empty string to disable
 Define.V2Module yes    # non-empty string to enable module name lookups (obsolete)
 Define.V2Name yes      # non-empty string to enable driver & card name lookups (obsolete)
 
-Include.libgen.File "/lib/generic.conf"  # private alsa-lib configuration
+# Include.libgen.File "/lib/generic.conf"  # private alsa-lib configuration
 
 If.driver {
        Condition {
```

**Comment by perexg** [comment]
As you can see, the ucm can have a private configuration for the PCM and control ALSA devices. It means that the global alsa-lib configuration is not used in this case. So yes, it's part of the UCM API now.

The private prefix which can be obtained using `_alibpref` key (https://github.com/alsa-project/alsa-lib/blob/0e4ba2ea8c0402f12a645032a14693eb9b1278e6/include/use-case.h#L261) may be removed by pulseaudio . Note that the number in the prefix may vary.

**Comment by agx** [comment]
@perexg so just to confirm: pa should just drop the prefix?

**Comment by perexg** [comment]
> @perexg so just to confirm: pa should just drop the prefix?

Yes, but only for the port name string. The unmodified device name must be used for the device open call.

**Comment by craftyguy** [comment]
this is fixed in pulseaudio 18.0
